leapsectz/leapsectz.go

```diff
@@ -20,102 +20,288 @@ limitations under the License.
 package leapsectz
 
 import (
+	"bytes"
 	"encoding/binary"
 	"errors"
 	"io"
+	"io/ioutil"
 	"os"
 	"time"
 )
 
 const file = "/usr/share/zoneinfo/right/UTC"
 
 var errBadData = errors.New("malformed time zone information")
+var errBadVersion = errors.New("version in file is not supported")
 
 // LeapSecond represents a leap second
 type LeapSecond struct {
-	Tleap uint32
-	Nleap uint32
+	Tleap uint64
+	Nleap int32
+}
+
+// Header represents file header structure. Fields names are copied from doc
+type Header struct {
+	// A four-octet unsigned integer specifying the number of UTC/local indicators contained in the body.
+	IsUtcCnt uint32
+	// A four-octet unsigned integer specifying the number of standard/wall indicators contained in the body.
+	IsStdCnt uint32
+	// A four-octet unsigned integer specifying the number of leap second records contained in the body.
+	LeapCnt uint32
+	// A four-octet unsigned integer specifying the number of transition times contained in the body.
+	TimeCnt uint32
+	// A four-octet unsigned integer specifying the number of local time type Records contained in the body - MUST NOT be zero.
+	TypeCnt uint32
+	// A four-octet unsigned integer specifying the total number of octets used by the set of time zone designations contained in the body.
+	CharCnt uint32
 }
 
 // Time returns when the leap second event occurs
 func (l LeapSecond) Time() time.Time {
```

Comment:
Can you unittest it? Seems very straightforward and will bump your coverage :)

Reply:
Yeah, done